x-pack/plugins/synthetics/public/legacy_uptime/components/fleet_package/browser/source_field.tsx

```diff
@@ -145,12 +143,134 @@ export const SourceField = ({
+            <CodeEditor
+              ariaLabel={i18n.translate(
+                'xpack.synthetics.createPackagePolicy.stepConfigure.requestBody.codeEditor.javascript.ariaLabel',
+                {
+                  defaultMessage: 'JavaScript code editor',
+                }
+              )}
+              id="javascript"
+              languageId={MonacoEditorLangId.JAVASCRIPT}
+              onChange={(code) => {
+                setConfig((prevConfig) => ({ ...prevConfig, inlineScript: code }));
+                onFieldBlur(ConfigKey.SOURCE_INLINE);
+              }}
+              value={config.inlineScript}
+            />
+          </EuiFormRow>
+          {params}
+        </>
+      ),
+    },
+    {
+      id: 'syntheticsBrowserScriptRecorderConfig',
+      name: (
+        <EuiFlexGroup responsive={false} alignItems="center" gutterSize="xs">
+          <EuiFlexItem grow={false}>
+            <FormattedMessage
+              id="xpack.synthetics.createPackagePolicy.stepConfigure.browser.scriptRecorder.label"
+              defaultMessage="Script recorder"
+            />
+          </EuiFlexItem>
+          <StyledBetaBadgeWrapper grow={false}>
+            <EuiBetaBadge
+              label={i18n.translate(
+                'xpack.synthetics.createPackagePolicy.stepConfigure.browser.scriptRecorder.experimentalLabel',
+                {
+                  defaultMessage: 'Tech preview',
+                }
+              )}
+              iconType="beaker"
+              tooltipContent={i18n.translate(
+                'xpack.synthetics.createPackagePolicy.stepConfigure.browser.scriptRecorder.experimentalTooltip',
+                {
+                  defaultMessage:
+                    'Preview the quickest way to create Elastic Synthetics monitoring scripts with our Elastic Synthetics Recorder',
+                }
+              )}
+            />
+          </StyledBetaBadgeWrapper>
+        </EuiFlexGroup>
+      ),
+      'data-test-subj': 'syntheticsSourceTab__scriptRecorder',
+      content: (
+        <>
+          <ScriptRecorderFields
+            onChange={({ scriptText, fileName }) =>
+              setConfig((prevConfig) => ({
+                ...prevConfig,
+                inlineScript: scriptText,
+                isGeneratedScript: true,
+                fileName,
+              }))
+            }
+            script={config.inlineScript}
+            fileName={config.fileName}
+          />
+          <EuiSpacer size="s" />
+          {params}
+        </>
+      ),
+    },
     {
       id: zipUrlSourceTabId,
       name: zipUrlLabel,
       'data-test-subj': `syntheticsSourceTab__zipUrl`,
       content: (
         <>
+          <EuiSpacer size="m" />
+          <EuiCallOut
+            title={
+              <FormattedMessage
+                id="xpack.synthetics.createPackagePolicy.stepConfigure.monitorIntegrationSettingsSection.browser.zipUrl.deprecation.title"
+                defaultMessage="Deprecated"
```

Comment:
```suggestion
                defaultMessage="Zip URL is deprecated"
```
Suggesting to make the title specific.